src/material-experimental/mdc-checkbox/harness/checkbox-harness.ts

```diff
@@ -21,12 +21,17 @@ export class MatCheckboxHarness extends ComponentHarness {
    * Gets a `HarnessPredicate` that can be used to search for a checkbox with specific attributes.
    * @param options Options for narrowing the search:
    *   - `label` finds a checkbox with specific label text.
+   *   - `name` finds a checkbox with specific name.
    * @return a `HarnessPredicate` configured with the given options.
    */
   static with(options: CheckboxHarnessFilters = {}): HarnessPredicate<MatCheckboxHarness> {
     return new HarnessPredicate(MatCheckboxHarness)
-        .addOption('label', options.label,
-            (harness, label) => HarnessPredicate.stringMatches(harness.getLabelText(), label));
+        .addOption(
+            'label', options.label,
+            (harness, label) => HarnessPredicate.stringMatches(harness.getLabelText(), label))
+        // We want to provide a filter option for "name" because there is no way for
```

Comment:
I'm not sure this comment fully explains the reason. It should mention that the name is moved to the internal native input element, so there's no good way for the user to get a reference to the parent mat-checkbox using the name, unless we give them this API.

Reply:
Good point. done